src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +138,312 @@ const ActorInit En_Pst_InitVars = {
+void EnPst_Talk(EnPst* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->msgEventScript, this->msgEventCallback, &this->msgEventArg4)) {
+        if (EnPst_HandleLetterDay1(this) != this->isLetterToKafeiDeposited) {
+            switch (gSaveContext.save.day) {
+                case 1:
+                    gSaveContext.save.weekEventReg[91] |= 4;
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BAA4.s")
+                case 2:
+                    if (EnPst_HandleLetterDay2(this)) {
+                        gSaveContext.save.weekEventReg[91] |= 8;
+                    } else {
+                        gSaveContext.save.weekEventReg[91] |= 4;
+                    }
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BBFC.s")
+                default:
+                    gSaveContext.save.weekEventReg[91] |= 8;
+                    break;
+            }
+        }
+        SubS_UpdateFlags(&this->stateFlags, 3, 7);
+        this->msgEventArg4 = 0;
+        this->actionFunc = EnPst_FollowSchedule;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BCF8.s")
+void EnPst_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnPst* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BD30.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 18.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gPostboxSkel, NULL, this->jointTable, this->morphTable,
+                       POSTBOX_LIMB_MAX);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    SubS_UpdateFlags(&this->stateFlags, 3, 7);
+    SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, 0);
+    this->actor.targetMode = 0;
+    Actor_SetScale(&this->actor, 0.02f);
+    this->actionFunc = EnPst_FollowSchedule;
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BD88.s")
+void EnPst_Destroy(Actor* thisx, PlayState* play) {
+    EnPst* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BD98.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2BE54.s")
+void EnPst_Update(Actor* thisx, PlayState* play) {
+    EnPst* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/EnPst_Init.s")
+    EnPst_CheckTalk(this, play);
+    this->actionFunc(this, play);
+    if (this->scheduleResult != POSTBOX_SCH_NONE) {
+        if (Actor_IsFacingPlayer(&this->actor, 0x1FFE)) {
+            this->unk214 = 0;
+            func_8013C964(&this->actor, play, 60.0f, 20.0f, 0, this->stateFlags & 7);
```

Comment:
oh hello, this is the first time I've seen a local actor's flags being passed (in part) to an outside function, that's interesting.